src/lib/y2partitioner/actions/delete_partition.rb

```diff
@@ -29,13 +29,94 @@ module Actions
     #
     # @see DeleteDevice
     class DeletePartition < DeleteDevice
+      # Confirmation message before performing the delete action
+      def confirm
+        if used_by_lvm?
+          confirm_for_used_by_lvm
+        elsif used_by_md_raid?
+          confirm_for_used_by_md_raid
+        else
+          super
+        end
+      end
+
       # Deletes the indicated partition (see {DeleteDevice#device})
       def delete
         log.info "deleting partition #{device}"
         disk_device = device.partitionable
         disk_device.partition_table.delete_partition(device)
         UIState.instance.select_row(disk_device)
       end
+
+    private
+
+      # Checks whether the partition is used as physical volume
+      #
+      # @return [Boolean] true if partition belongs to a volume group; false otherwise
+      def used_by_lvm?
+        !vg.nil?
+      end
+
+      # Volume group that the partition belongs to
+      #
+      # @return [Y2Storage::LvmVg, nil] nil if the partition does not belong to
+      #   a volume group
+      def vg
+        device.descendants.find { |d| d.is?(:lvm_vg) }
+      end
+
+      # Checks whether the partition is used by a md raid
+      #
+      # @return [Boolean] true if partition belongs to a md raid; false otherwise
+      def used_by_md_raid?
+        !md_raid.nil?
+      end
+
+      # Md Raid that the partition belongs to
+      #
+      # @return [Y2Storage::Md, nil] nil if the partition does not belong to a md raid
+      def md_raid
```

Comment:
Relatively similar to the former, we maybe should move this to `BlkDevice` with a good name.